### PR TITLE
[RATOM-134] add arbitrary labels

### PR DESCRIPTION
**Pull request by michael-caktus**
forked the PUT handler on message_detail view to handle adding labels.
The UniqueTogether constraint on the label model resulted in some interesting problems during validation, as mentioned [in this medium article](https://medium.com/django-rest-framework/dealing-with-unique-constraints-in-nested-serializers-dade33b831d9). Instead of overriding all of the validators and possibly losing some desired functionality, we've simple performed the get_or_create in the view.

Tested with integration testing on the front end.

### CODE REVIEW

File: api/views/message.py [hunk 1]
```diff
@@ -38,11 +38,24 @@ def message_detail(request, pk):
...
         We don't really edit messages-- this endpoint updates an associated MessageAudit
         """
-        serialized_audit = MessageAuditSerializer(message.audit, data=request.data)
-        if serialized_audit.is_valid():
-            serialized_audit.save(updated_by=request.user)
-            return Response(serialized_audit.data, status=status.HTTP_201_CREATED)
-        return Response(serialized_audit.errors, status=status.HTTP_400_BAD_REQUEST)
+        if "labels" in request.data:
+            for label in request.data["labels"]:
+                (lb, _) = Label.objects.get_or_create(**label)
```
Comment (JeremyGibson) [inline api/views/message.py]: the parens are not necessary

File: api/views/message.py [hunk 2]
```diff
@@ -38,11 +38,24 @@ def message_detail(request, pk):
         """
         We don't really edit messages-- this endpoint updates an associated MessageAudit
         """
-        serialized_audit = MessageAuditSerializer(message.audit, data=request.data)
-        if serialized_audit.is_valid():
-            serialized_audit.save(updated_by=request.user)
-            return Response(serialized_audit.data, status=status.HTTP_201_CREATED)
-        return Response(serialized_audit.errors, status=status.HTTP_400_BAD_REQUEST)
+        if "labels" in request.data:
+            for label in request.data["labels"]:
```
Comment (JeremyGibson) [inline api/views/message.py]: Is this for bulk actions later? Would this ever have more than one label?

Comment (michael-caktus) [reply api/views/message.py]: Yeah, I can change that-- it's an artifact of an earlier implementation.

File: api/views/message.py (hunk 1 above)
Comment (michael-caktus) [reply api/views/message.py]: removed em!

**Review by JeremyGibson — APPROVED** [review]
See what you think of this change, If you are uncomfortable with it roll back and :shipit:

**Pull request merged**